packages/amplify-category-api/commands/api/push.js

```diff
@@ -9,8 +9,7 @@ module.exports = {
     context.amplify.constructExeInfo(context);
     return amplify.pushResources(context, category, resourceName)
       .catch((err) => {
-        context.print.info(err.stack);
-        context.print.error('There was an error pushing the API resource');
+        context.print.error(err.toString());
```

Comment:
wouldn't the error message be more verbose and unformatted if we just do a toString()?
 Also, we should retain this error message - context.print.error('There was an error pushing the API resource');

Reply:
It makes the error a lot more readable in my opinion. The stack trace is only marginally helpful to the user and now you get much cleaner, more helpful error messages. I have added back the original error text before the new text but in general I think its better to allow the origin of the error to define what type of error it is instead of having the catch all text hard coded into the handler. In this instance, the transformer has not tried to push yet so the text "There was an error pushing the API resource" isn't as focused as it could be. Ideally each error type would know how to represent itself and then the downstream code can throw one of those errors and the it will bubble up to the top level where it can be nicely handled (by switching on an enum for example).